bundles/http/Citrix/AdobeSign/31fff7a5-ffa8-4019-9e2f-e477837f92de.js

```diff
@@ -0,0 +1,594 @@
+    const userids = await syncUsers(dataStore, client, integrationParameters,);
+    const agreements = await syncAgreements(dataStore, client, userids, integrationParameters, latestSynchronizationTime);
+    await syncArgreementDetails(client, dataStore, agreements)
+}
+
+async function libraryDocumentView({ client, dataStore, actionParameters }) {
+    const response = await client.fetch(`/api/rest/v6/libraryDocuments/${actionParameters.libraryDocumentId}/views`, {
+        method: 'POST',
+        headers: {
+            "x-api-user": `email:${actionParameters.email}`,
+        },
+        body: JSON.stringify({
+            name: actionParameters.name,
+            commonViewConfiguration: {
+                autoLoginUser: true
+            }
+        })
+    })
+    const json = await response.json()
+    if (!response.ok) {
+        throw new Error(JSON.stringify(json))
+    }
+    const libraryViewList = json['libraryViewList'] ?? []
+    let data = libraryViewList.map(libraryView => {
+        return {
+            "libraryDocumentId": actionParameters.libraryDocumentId,
+            "email": actionParameters.email,
+            "url": libraryView.url,
+            "embeddedCode": libraryView.embeddedCode,
+            "name": libraryView.name,
+            "isCurrent": libraryView.isCurrent
+        }
+    })
+    dataStore.save("libraryDocumentViews", data)
+}
+
+async function agreementview({ client, dataStore, actionParameters }) {
+    const response = await client.fetch(`/api/rest/v6/agreements/${actionParameters.agreementId}/views`, {
+        method: 'POST',
+        headers: {
+            "x-api-user": `email:${actionParameters.email}`,
+        },
+        body: JSON.stringify({
+            name: actionParameters.name,
+            commonViewConfiguration: {
+                autoLoginUser: true
+            }
+        })
+    })
+    const json = await response.json()
+    if (!response.ok) {
+        throw new Error(JSON.stringify(json))
+    }
+    const agreementViewList = json['agreementViewList'] ?? []
+    let data = agreementViewList.map(agreementView => {
+        return {
+            "agreement_id": actionParameters.agreementId,
+            "email": actionParameters.email,
+            "url": agreementView.url,
+            "embeddedCode": agreementView.embeddedCode,
+            "expiration": agreementView.expiration,
+            "name": agreementView.name,
+            "isCurrent": agreementView.isCurrent
+        }
+    })
+    dataStore.save("agreementViews", data)
+}
+
+async function sendAgreement({ dataStore, client, actionParameters, integrationParameters }) {
+    let transientDocuments = null
+    let fileInfos = {
+        "libraryDocumentId": `${actionParameters.libraryDocumentId}`
+    }
+    if (actionParameters.libraryDocumentId == null) {
+        let transientDocument = await uploadFiles(dataStore, client, actionParameters)
+        if(Boolean(transientDocument) === false){
```

Comment:
Now I totally don't understand why are you doing this.

Before you returned `false`(boolean) when there was an error. And `number` when it was ok.
You could easily distinguish what is an error and what is not.
I suggested to do just `if(transientDocument === false)`

But no, you "fixed" it to return only `number` and then convert it to boolean. Why?
If you think that it is better to return just a number, then you can do `if (transientDocument === 0){` but why to convert it again to boolean?

Ok. Here's how you should do it:
On line 375 just `throw new Error('File not selected')`
Then you can delete lines 262-264.
And everything will simply work as before.

Reply:
Changes Implemented

Reply:
Changes Implemented